catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -0,0 +1,312 @@
+package ddf.catalog.history;
+
+import static ddf.catalog.core.versioning.MetacardVersion.HISTORY_METACARDS_PROPERTY;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.apache.shiro.SecurityUtils;
+import org.codice.ddf.security.common.Security;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.io.ByteSource;
+import com.google.common.io.Files;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.content.StorageProvider;
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.content.data.impl.ContentItemImpl;
+import ddf.catalog.content.operation.CreateStorageRequest;
+import ddf.catalog.content.operation.CreateStorageResponse;
+import ddf.catalog.content.operation.UpdateStorageResponse;
+import ddf.catalog.content.operation.impl.CreateStorageRequestImpl;
+import ddf.catalog.core.versioning.MetacardVersion;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.CreateResponse;
+import ddf.catalog.operation.Update;
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.CreateResponseImpl;
+import ddf.catalog.operation.impl.UpdateResponseImpl;
+import ddf.catalog.source.IngestException;
+import ddf.security.Subject;
+
+public class Historian {
+    public static final String ALREADY_VERSIONED = "already-versioned";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Historian.class);
+
+    private static final String HISTORY_METACARDS = "history-metacards";
+
+    private boolean historyEnabled = true;
+
+    public Historian() {
+    }
+
+    private final Cache<String, List<Callable<Boolean>>> staged = CacheBuilder.newBuilder()
+            .expireAfterAccess(1, TimeUnit.HOURS)
+            .build();
+
+    public CreateRequest versionCreate(CreateRequest createRequest) {
+        if (!historyEnabled) {
+            return createRequest;
+        }
+        if (createRequest.getProperties()
+                .getOrDefault(ALREADY_VERSIONED, "")
+                .equals(true)) {
```

Comment:
Your default value is an empty string whereas your expected value is a boolean, then you're doing an equals call to a boolean. Why not just...
```java
if (createRequest.getProperties().getOrDefault(ALREADY_VERSIONED, false)) {
...
```